06-building-a-program-start-to-finish/solving-the-problem/Output/model/inverted-index.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * The Inverted Index class
+ */
+class InvertedIndex {
+
+  /**
+   * Constructor initializes indexes as an empty object
+   * and keeps track of the files that have been indexed, their titles and their
+   * indices in the respective files
+   * @constructor
+   */
+  constructor() {
+    this.allIndexed = {};
+    this.allTitles = {};
+    this.documentsIndex = {};
+  }
+
+  /**
+   * Create Index
+   * createIndex method takes a single document and builds an index from it
+   * @param {Object} fileContent - the file content to be indexed
+   * @param {String} fileName - the name of the file to be indexed
+   * @return {Object} all words in the file and their corresponding indexes
+   */
+  createIndex(fileContent, fileName) {
+    const indexed = {};
+    const titles = [];
+    fileContent.forEach((book, index) => {
+      let text = `${book.title} ${book.text}`;
+      titles.push(book.title);
+      text = this.tokenize(text);
+      text.forEach((word) => {
+        if (!(word in indexed)) {
+          indexed[word] = [(index)];
+        } else if (!(indexed[word].includes(index))) {
+          indexed[word].push(index);
+        }
+      });
+    });
+    const documents = [];
+    for (let i = 0; i < titles.length; i += 1) {
+      documents.push(i);
+    }
+    this.allIndexed[fileName] = indexed;
+    this.allTitles[fileName] = titles;
+    this.documentsIndex[fileName] = documents;
+    return indexed;
+  }
+
+  /**
+   * Get Index
+   * getIndex method gets the indexed file with words from documents that were
+   * found. If the file has not been indexed, it calls the create index method
+   * @param {Object} fileContent - the file content we want to get indexed
+   * @param {String} filename - the name of the file to be indexed
+   * @return {Object} all words in the file and their corresponding indexes
+   */
+  getIndex(fileContent, fileName) {
+    if (fileName in this.allIndexed) {
+      return this.allIndexed[fileName];
+    }
+    return this.createIndex(fileContent, fileName);
+  }
```

Comment:
Why are you creating your Index here? It seems as though you didn't follow the instruction in the description of [Checkpoint 1](https://docs.google.com/document/d/1Y1171vb09G-KhvHafgK219vxf9bhfgkUBA_CxbGPAGU/edit) for your getIndex()